File: library/src/main/java/su/levenetc/android/textsurface/animations/Just.java
package su.levenetc.android.textsurface.animations;

import android.support.annotation.Nullable;

import su.levenetc.android.textsurface.contants.TYPE;
import su.levenetc.android.textsurface.Text;
import su.levenetc.android.textsurface.interfaces.IEndListener;

/**
 * Created by Eugene Levenetc.
 */
public class Just extends AbstractSurfaceAnimation {

	public static AnimationsSet show(Text... texts) {
		Just[] justs = new Just[texts.length];
		for (int i = 0; i < texts.length; i++) justs[i] = Just.show(texts[i]);
		return new AnimationsSet(TYPE.PARALLEL, justs);
	}

	public static Just show(Text text) {
		return new Just(text);
	}

	private Just(Text text) {
		super(text, 0);
	}

	@Override public void start(@Nullable IEndListener listener) {
		textSurface.invalidate();
		if (listener != null) listener.onAnimationEnd(this);
	}
}

File: library/src/main/java/su/levenetc/android/textsurface/animations/Just.kt
package su.levenetc.android.textsurface.animations

import su.levenetc.android.textsurface.Text
import su.levenetc.android.textsurface.TextSurface
import su.levenetc.android.textsurface.contants.TYPE
import su.levenetc.android.textsurface.interfaces.IEndListener

/**
 * Created by Eugene Levenetc.
 */
class Just private constructor(text: Text) : AbstractSurfaceAnimation(text, 0) {
    //TODO: check unused textSurface
    override lateinit var textSurface: TextSurface

    override fun start(listener: IEndListener) {
        textSurface!!.invalidate()
        listener?.onAnimationEnd(this)
    }

    companion object {
        fun show(vararg texts: Text): AnimationsSet {
            val justs = arrayOfNulls<Just>(texts.size)
            for (i in texts.indices) justs[i] = show(texts[i])
            return AnimationsSet(TYPE.PARALLEL, *justs)
        }

        fun show(text: Text): Just {
            return Just(text)
        }
    }
}